src/main/cpp/blaze.cc

```diff
@@ -984,10 +984,11 @@ static bool IsVolatileArg(const string &arg) {
 }
 
 // Returns true if the server needs to be restarted to accommodate changes
-// between the two argument lists.
+// between the two argument lists. Populates different_options with the options that differ.
 static bool AreStartupOptionsDifferent(
     const vector<string> &running_server_args,
-    const vector<string> &requested_args) {
+    const vector<string> &requested_args,
+    vector<string> *different_options) {
```

Comment:
I would consider passing in two lists here, so that all old options can be printed together and all new options can be printed together.

For example, if I am running:
```
$ bazel --foo=42 --bar --zoo build ...
$ bazel -zoo --foo=23 --goo build ...
```
I would like to see a message like this:
```
Running Bazel server needs to be killed, because the following startup options are different:
  - Only in running server: --foo=42 --bar
  - Only in new server: --foo=23 --goo
```

What do you think?

Reply:
That's even better. I'll make the change.